tests/unit/test_charm.py

```diff
@@ -425,6 +425,37 @@ def test_add_admin_user(self, mock_exec):
             timeout=60,
         )
 
+    @patch.object(Container, "exec")
+    def test_anonymize_user(self, mock_exec):
```

Comment:
Should this also have an integration test?

Reply:
It seems overkill to me given criticallity

Reply:
Not so much about the patch, it is more about that the action being added works on a real version of the charm

Reply:
It's unit tested so I'm assuming Discourse's contract is fine (once the changes land). It has been manually tested, oc.

Reply:
I think you are referring to the PR that added the add-admin or similar action. The difference on that one was there was an integration test and the conversation was about to degree to which the final state should be checked, in this case there wasn't an integration test so the cases are different.

Also, anonymizing user data I assume is in response to a user request that is covered by regulation. Perhaps that warrants some thinking about business criticality compared to an action that adds an admin user which is probably not covered by regulation as much?